storage/sectorstorage/ffiwrapper/basicfs/fs.go

```diff
@@ -0,0 +1,70 @@
+package basicfs
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+
+	"github.com/filecoin-project/lotus/storage/sectorstorage/stores"
+)
+
+type sectorFile struct {
+	abi.SectorID
+	stores.SectorFileType
+}
+
+type Provider struct {
+	Root string
+
+	lk         sync.Mutex
+	waitSector map[sectorFile]chan struct{}
+}
+
+func (b *Provider) AcquireSector(ctx context.Context, id abi.SectorID, existing stores.SectorFileType, allocate stores.SectorFileType, sealing bool) (stores.SectorPaths, func(), error) {
+	os.Mkdir(filepath.Join(b.Root, stores.FTUnsealed.String()), 0755)
```

Comment:
I would prefer we check these errors, ignore 'already exists' errors, but permissions and other fun things should get floated back up